drivers/interrupt_controller/intc_plic.c

```diff
@@ -109,9 +109,9 @@ static inline uint32_t get_plic_enabled_size(const struct device *dev)
 	return local_irq_to_reg_index(config->num_irqs) + 1;
 }
 
-static inline uint32_t get_first_context(uint32_t hartid)
+static ALWAYS_INLINE uint32_t get_hart_context(uint32_t hartid, uint32_t nr_contexts, uint32_t context)
```

Comment:
we need a way to configure the `nr_contexts`

Reply:
Hey @ycsin, I am stuck in the middle of trying to bring Zephyr up in supervisor mode, I can try and get around to testing this out tomorrow

Reply:
> Hey @ycsin, I am stuck in the middle of trying to bring Zephyr up in supervisor mode, I can try and get around to testing this out tomorrow

No worries, good luck with the bring-up